opencog/haskell/TruthValue_CWrapper.cpp

```diff
@@ -0,0 +1,107 @@
+#include "TruthValue_CWrapper.h"
+
+
+int TruthValue_toRawType(TruthValuePtr tv
+                        , char** tv_type
+                        , double* parameters)
+{
+    const std::string & tvt = classserver().getTypeName(tv->getType());
+
+    *tv_type = (char*) malloc(sizeof(char) * (tvt.length()+1));
+    if(! *tv_type)
+        throw RuntimeException(TRACE_INFO,"Failed malloc.");
+    std::strcpy(*tv_type, tvt.c_str());
+
+    if (strcmp(*tv_type,"SimpleTruthValue") == 0){
```

Comment:
Yea I thought a bit about supporting all kinds of values but wasn't quite sure on how to best do that right now. So I decide to just change the minimal amount needed to get it to work.
But I am a bit more comfortable with that now so will implement that next.